README.md

```diff
@@ -407,6 +407,165 @@ after a sequence of diffs are handled.
+* `toMap` _Object_: an object whose keys are the IDs of the target documents. The values are the last versions requested of each document (non-inclusive). For example, `{abc: 3}` will fetch ops for document with ID `abc` up to version `3` (_not_ inclusive)
+* `options`: _Object (optional)_: options can be passed directly to the database driver's `getOpsBulk` inside the `opsOptions` property: `{opsOptions: {metadata: true}}`
+* `callback`: _Function_: a callback with the signature `function (error: Error, opsMap: Object): void;`, where `opsMap` is an object whose keys are the IDs of the requested documents, and their values are the arrays of requested ops, eg `{abc: []}`
+
+#### `fetch`
+
+```javascript
+backend.fetch(agent: Agent, index: string, id: string, options: Object, callback: Function): void;
+```
+
+Fetch the current snapshot of a document.
+
+* `agent` _Agent_: connection agent to pass to the middleware
+* `index` _string_: the name of the target collection or projection
+* `id` _string_: the document ID
+* `options`: _Object (optional)_: options can be passed directly to the database driver's `fetch` inside the `snapshotOptions` property: `{snapshotOptions: {metadata: true}}`
+* `callback`: _Function_: a callback with the signature `function (error: Error, snapshot: Snapshot): void;`, where `snapshot` is the requested snapshot
+
+#### `fetchBulk`
+
+```javascript
+backend.fetchBulk(agent: Agent, index: string, ids: string[], options: Object, callback: Function): void;
+```
+
+Fetch multiple document snapshots from a collection.
+
+* `agent` _Agent_: connection agent to pass to the middleware
+* `index` _string_: the name of the target collection or projection
+* `ids` _string[]_: array of document IDs
+* `options`: _Object (optional)_: options can be passed directly to the database driver's `fetchBulk` inside the `snapshotOptions` property: `{snapshotOptions: {metadata: true}}`
+* `callback`: _Function_: a callback with the signature `function (error: Error, snapshotMap: Object): void;`, where `snapshotMap` is an object whose keys are the requested IDs, and the values are the requested `Snapshot`s
+
+<!-- TODO: Is it okay to have ignored `subscribe`, `subscribeBulk`, `querySubscribe`? -->
+
+<!-- TODO: Should we ignore `queryFetch`? -->
+
+#### `queryFetch`
+
+```javascript
+backend.queryFetch(agent: Agent, index: string, query: Object, options: Object, callback: Function): void;
+```
+
+Fetch snapshots that match the provided query.
+
+* `agent` _Agent_: connection agent to pass to the middleware
+* `index` _string_: the name of the target collection or projection
+* `query` _Object_: a query object, whose format will depend on the database adapter being used
+<!-- TODO: I don't think options are used? -->
```

Comment:
These are the same as `.submit` - mostly there for future-proofing, so that options can be sent through to the driver.